### PR TITLE
Add dockerfile for the OCP CI build image

### PR DESCRIPTION
**Pull request by fedepaol**
# Description

Openshift CI allows to check out our repo in a src image, optionally provided by a
Dockerfile.
We have an issue right now (described here https://github.com/openshift/release/pull/6395 ) because the oc command shipped with the image provided by the ocp CI does not support applying all the files in a given folder.
With this PR we provide a dockerfile that can be used to produce an image in place of the one provided by the CI.

Fixes # (issue)

## Type of change

Please select the appropiate options:

- [c ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Testing

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:

- Versions:
- Hardware:

## Checklist

- [ x ] I have performed a self-review of my own code
- [ ] If a change is adding a feature, it should require a change to the README.md and the review should catch this.
- [ ] If the change is a fix, it should have an issue. The review should make sure the comments state the issue (not just the number) and it should use the keywords that will close the issue on merge.
- [ ] A change should not being merged unless it passes CI or there is a comment/update saying what testing was passed.
- [ x ] PRs should not be merged unless positively reviewed.

### CODE REVIEW

File: features/Dockefile.buildimage [hunk 1]
```diff
@@ -0,0 +1,25 @@
+FROM fedora:30 as build-tools
```
Comment (e-minguez) [inline features/Dockefile.buildimage]: Any particular reason of using fedora:30 instead fedora:31? also the `as build-tools` I think it is not needed as it is not a multistage build (there is no FROM build-tools later)

Comment (fedepaol) [reply features/Dockefile.buildimage]: None, a part the fact that I "took inspiration from" a Dockerfile built like that.
Fixing.

File: features/Dockefile.buildimage [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ENV GOPATH /go
+ENV GOBIN /go/bin
+ENV GOCACHE /go/.cache
+ENV PATH=$PATH:$GOPATH/bin
+
+# rpms required for building and running test suites
+RUN dnf -y install \
+    golang-1.12.13-1.fc30 \
```
Comment (e-minguez) [inline features/Dockefile.buildimage]: is this golang specific version needed?
If so, what about having an environment variable instead?

Comment (fedepaol) [reply features/Dockefile.buildimage]: Nope, but on the long term we want to use the same Dockerfile for local builds.
By pinning a specific version we make sure it won't change over time and we get reproducible builds.

Ok for the env variable.

File: features/Dockefile.buildimage (hunk 1 above)
Comment (fedepaol) [reply features/Dockefile.buildimage]: Done

File: features/Dockefile.buildimage (hunk 2 above)
Comment (fedepaol) [reply features/Dockefile.buildimage]: Done

File: features/Dockefile.buildimage [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    && dnf clean all
+
+# get required golang tools and OC client
+RUN go get github.com/onsi/ginkgo/ginkgo && \
+    go get github.com/onsi/gomega/... && \
+    go get -u golang.org/x/lint/golint && \
+    export latest_oc_client_version=$(curl https://mirror.openshift.com/pub/openshift-v4/clients/ocp/latest/ 2>/dev/null | grep -o \"openshift-client-linux-4.*tar.gz\" | tr -d \") && \
+    curl -JL https://mirror.openshift.com/pub/openshift-v4/clients/ocp/latest/${latest_oc_client_version} -o oc.tar.gz && \
+    tar -xzvf oc.tar.gz && \
+    mv oc /usr/bin/oc
```
Comment (e-minguez) [inline features/Dockefile.buildimage]: nit: non packaged binaries should go into /usr/local/bin/ (for system wide binaries) or /usr/local/sbin (for root only binaries)

Comment (e-minguez) [reply features/Dockefile.buildimage]: Also, removing the oc.tar.gz would be nice:

```
mv oc /usr/local/bin/oc && \
rm -f oc.tar.gz
```

Comment (fedepaol) [reply features/Dockefile.buildimage]: Will do, thanks!

Comment (fedepaol) [reply features/Dockefile.buildimage]: Done

**Comment by fedepaol** [comment]
@e-minguez you need also to put a lgtm label

**Pull request merged**